### PR TITLE
feat: add support for VNI in instance module

### PR DESCRIPTION
**Pull request by Aashiq-J**
### Description

Issue: https://github.ibm.com/GoldenEye/issues/issues/11293

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [x] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: modules/instance/main.tf
Comment (rajatagarwal-ibm) [inline modules/instance/main.tf:1]: There seems to be too many validation inside the "resource" block. Would you like to move them to the locals?

File: modules/instance/variables.tf [hunk 1]
```diff
@@ -40,7 +40,13 @@ variable "primary_network_interface" {
     interface_name       = string
     security_groups      = list(string)
     primary_ipv4_address = string
+    allow_ip_spoofing    = optional(bool)
   }))
+
+  validation {
+    condition     = var.no_of_instances == length(var.primary_network_interface)
```
Comment (rajatagarwal-ibm) [inline modules/instance/variables.tf:47]: I think `no_of_instances` variable should be renamed to `number_of_instances` for readability.

File: modules/instance/main.tf [hunk 2]
```diff
@@ -1,46 +1,117 @@
...
+      address = var.primary_network_interface[count.index].primary_ipv4_address
+    }
+  }
+}
+
+resource "ibm_is_virtual_network_interface" "additional_vnis" {
+  count                     = length(var.network_interfaces) > 0 && !var.use_legacy_network_interface ? length(var.network_interfaces) : 0
+  name                      = var.network_interfaces[count.index].interface_name
+  subnet                    = var.network_interfaces[count.index].subnet
+  allow_ip_spoofing         = var.network_interfaces[count.index].allow_ip_spoofing
```
Comment (rajatagarwal-ibm) [inline modules/instance/main.tf:25]: `allow_ip_spoofing` is optional. This piece of code migh fail without any validation around it

File: modules/instance/main.tf
Comment (rajatagarwal-ibm) [reply modules/instance/main.tf:1]: I don't see validation which make sure that if user selects legacy NI, then `allow_ip_spoofing` should not be set, unless legacy handles that?

**Review by rajatagarwal-ibm — CHANGES_REQUESTED** [review]
Have you done the following testing? 

- [ ] Creating VNI with default settings.
- [ ] Creating VNI with `allow_ip_spoofiing` value set.
- [ ] Creating legacy NI

File: modules/instance/main.tf (hunk 2 above)
Comment (Aashiq-J) [reply modules/instance/main.tf:25]: Validation not required. I have tested it, if no value is set it is set to false.

**Review by maheshwarishikha — DISMISSED** [review]

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**